src/Shell/Task/ModelTask.php

```diff
@@ -273,10 +273,17 @@ public function findBelongsTo($model, array $associations)
                         $tmpModelName = Inflector::camelize($found);
                     }
                 }
+                $foreignKeys = (array) TableRegistry::get($tmpModelName)->primaryKey();
+                foreach ($foreignKeys as $k => $v) {
+                    if ($v === 'id') {
```

Comment:
TLDR
In comments bekongTo assoc def,
before: 'foreignKey' => 'article_id'
after: 'foreignKey' => ['site_id', 'article_id']

Emperically working purpose of this part is to get the full composite primary key list of fields of the foreign table. However if you are in comments and fieldName is 'article_id' $foreignKeys will yield a list of 'id, tenant_id' instead of 'article_id, tenant_id' - it is latter that - assumed my convention - the assoc foreign key list needs to look like from the comments's point of view.

It would, however be best if first of all the defined FKs in the sql schema wiuld be respected and only if none where set a convention based hack / assumtion (like above) was in place.

Don't know how to do that.
Don't know if the assumption made is good enough.